### DISCUSSION TITLE
Multiple levels of debug logging?

### DISCUSSION THREAD

**Discussion by ramnes** [qtile/qtile#2408] [Ideas]
I feel like the debug logging level logs way too much stuff. Every time I use it, I have a very hard time gathering useful information because of how verbose it is. What would you think of adding new debug levels, for example a "verbose" level and logger function?

**Comment by m-col** [comment]
Same, I always find myself using `logger.info` when debugging changes I make. Looks like `logger.info` is only used 29 times in libqtile. Maybe can upgrade some messages from debug to info?

**Reply by ramnes** [reply]
Doesn't seem really intuitive to me to use the info log level for debugging. :)

I think we should rather move most of the info stuff in debug, and some of the warnings as info, so that warnings really are warnings, and info really is info, not debug.

**Reply by tych0** [reply]
Are there debug logs which are particularly spammy? Perhaps we should try to curate debug logs vs. just declaring that it is unusable and trying again.

**Reply by ramnes** [reply]
I think we need to do both. For example, a single X event can trigger up to four lines of logs:

```
2021-04-21 14:52:32,995 DEBUG libqtile core.py:_xpoll():L276 PropertyNotify
2021-04-21 14:52:32,996 DEBUG libqtile core.py:_xpoll():L279 Handling: PropertyNotify
2021-04-21 14:52:32,996 DEBUG libqtile window.py:handle_PropertyNotify():L1244 PropertyNotifyEvent: _NET_WM_USER_TIME
2021-04-21 14:52:32,996 INFO libqtile window.py:handle_PropertyNotify():L1280 Unknown window property: _NET_WM_USER_TIME
```

We probably only want to keep line 2 and 4 but still, I think that line 4 should be DEBUG instead of INFO, and that line 2 should be put aside for the verbose debug mode.

**Reply by tych0** [reply]
I don't know, if even maintainers don't want to see the debug log, perhaps we should just delete lines 1-3 and change 4 to debug?

**Reply by ramnes** [reply]
I think I do want to see both line 2 and 4, it's just that I want these two lines in different debugging scenarios. The thing I'm not sure about is which one I want to see more often than the other.

**Comment by m-col** [comment]
Yeah fair, that too!

**Reply by m-col** [reply]
Huh, replying to discussions via email doesn't send it to the right place.... can't say I'm surprised, github

**Comment by ramnes** [comment]
Related PR: #2416
